src/main.rs

```diff
@@ -826,6 +855,13 @@ fn main() -> ExitCode {
         || match main_impl(std::env::args_os(), Some(global_logger)) {
             Ok(_) => Ok(ExitCode::SUCCESS),
             Err(err) => {
+                // NB: downcast of error is discouraged
+                // and we need to do it more we should look into more structured
+                // custom errors for our code base, like the `thiserror` crate.
+                // https://www.reddit.com/r/learnrust/comments/zovj2x/how_to_match_on_underlying_error_when_using_anyhow/
+                if let Some(ExitSilently) = err.downcast_ref() {
+                    return Ok(ExitCode::FAILURE);
+                }
```

Comment:
I agree that `anyhow` was simplifying structured errors too much. Though, in this case, can you extend `main_impl` to return `Result<ExitCode>` instead?